serve/graph/model/cursor.go

```diff
@@ -0,0 +1,101 @@
+package model
+
+import (
+	"encoding/base64"
+	"errors"
+	"strconv"
+	"strings"
+)
+
+// Cursor represents a string key of an element position in a sequential list of edges.
+type Cursor string
+
+// Create a cursor by base64-encoding the ID.
+func NewCursor(id string) Cursor {
+	return Cursor(base64.StdEncoding.EncodeToString([]byte(id)))
+}
+
+// UnmarshalGraphQL unmarshal incoming cursor into a local variable.
+func (c *Cursor) UnmarshalGraphQL(input interface{}) error {
+	var err error
+
+	switch input := input.(type) {
+	case string:
+		*c = Cursor(input)
+	case int32:
```

Comment:
what if a cursor is an int? this code is a bit flaky, making it difficult to keep it compatible with new types, like int.